### PR TITLE
[NEW] In-app notification

### LINKED ISSUE #545: [NEW] In-app notifications
Subscribe to socket notifications and raise local notifications

### PR DESCRIPTION
**Pull request by diegolmello**
@RocketChat/ReactNative

Closes #545

image

### CODE REVIEW

**Comment by ggazzo** [comment]
I'm not sure about the notification IF we are on main screen AND using unread on top

**Comment by ggazzo** [comment]
good job btw :)

**Comment by diegolmello** [comment]
Thanks to @pranavpandey1998official. He did most of the hard work :)

File: app/notifications/inApp/index.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	}
+
+	goToRoom = async() => {
+		const { notification: { payload }, navigation } = this.props;
+		const { rid, type, prid } = payload;
+		if (!rid) {
+			return;
+		}
+		const name = type === 'p' ? payload.name : payload.sender.username;
+		await navigation.navigate('RoomsListView');
```
Comment (pranavpandey1998official) [inline app/notifications/inApp/index.js]: nice work around

**Comment by pranavpandey1998official** [comment]
It still overlaps with status bar if you look closely

**Comment by diegolmello** [comment]
> It still overlaps with status bar if you look closely

Please post a screenshot.

**Comment by diegolmello** [comment]
> It still overlaps with status bar if you look closely

@pranavpandey1998official Please post a screenshot.

**Pull request merged**